### PR TITLE
Fix overriden base container name not used to fetch logs on KPO trigger_reentry

### PR DESCRIPTION
**Pull request by peloyeje**
Ensures that self.pod_manager.fetch_container_logs use instance-level self.base_container_name instead of BASE_CONTAINER_NAME.
Fixes log fetching when base container name is provided
Also attempts at simplifying the assignment logic for self.container_logs in the init

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by peloyeje** [comment]
Will fix static checks asap

**Comment by peloyeje** [comment]
Static checks fixed

**Review by romsharon98 — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/operators/pod.py [hunk 1]
```diff
@@ -353,9 +352,9 @@ def __init__(
         self.cluster_context = cluster_context
         self.reattach_on_restart = reattach_on_restart
         self.get_logs = get_logs
-        self.container_logs = container_logs
-        if self.container_logs == KubernetesPodOperator.BASE_CONTAINER_NAME:
-            self.container_logs = base_container_name or self.BASE_CONTAINER_NAME
+        self.base_container_name = base_container_name or self.BASE_CONTAINER_NAME
```
Comment (hussein-awala) [inline airflow/providers/cncf/kubernetes/operators/pod.py]: You can change the default value of `base_container_name` to `BASE_CONTAINER_NAME` and remove this line

**Review by hussein-awala — APPROVED** [review]
I added a small nit, otherwise LGTM.

File: airflow/providers/cncf/kubernetes/operators/pod.py (hunk 1 above)
Comment (eladkal) [reply airflow/providers/cncf/kubernetes/operators/pod.py]: @peloyeje can you handle the above comment?

Comment (peloyeje) [reply airflow/providers/cncf/kubernetes/operators/pod.py]: Done, sorry for the delay.
Thanks for the review @hussein-awala!

**Pull request closed without merging**